cmake/alpakaCommon.cmake

```diff
@@ -328,6 +328,23 @@ if(NOT alpaka_HAS_STD_MATH_CONSTANTS)
     target_compile_definitions(alpaka INTERFACE "$<$<OR:$<CXX_COMPILER_ID:MSVC>,$<AND:$<COMPILE_LANGUAGE:CUDA>,$<PLATFORM_ID:Windows>>>:_USE_MATH_DEFINES>")
 endif()
 
+#-------------------------------------------------------------------------------
+# Include mdspan
+
+set(alpaka_USE_MDSPAN "INTERNAL" CACHE STRING "Use std::mdspan with alpaka")
+set_property(CACHE alpaka_USE_MDSPAN PROPERTY STRINGS "SYSTEM;OFF")
+
+if (alpaka_USE_MDSPAN STREQUAL "SYSTEM")
+    find_package(mdspan REQUIRED)
+    target_link_libraries(alpaka INTERFACE std::mdspan)
+    target_compile_definitions(alpaka INTERFACE ALPAKA_USE_MDSPAN)
+elseif (alpaka_USE_MDSPAN STREQUAL "INTERNAL")
+    # this is handled in thirdParty/CMakeLists.txt
+elseif (alpaka_USE_MDSPAN STREQUAL "OFF")
+else ()
+    message(FATAL_ERROR "Invalid option for alpaka_USE_MDSPAN")
+endif()
+
```

Comment:
You should add the compile definition here, because `thirdparty/CMakeLists.txt` is not installed. So if you install alpaka and use it in another project, the compiler definition `ALPAKA_USE_MDSPAN` is not set.

```cmake
+if (NOT alpaka_USE_MDSPAN STREQUAL "OFF")
+    target_compile_definitions(alpaka INTERFACE ALPAKA_USE_MDSPAN)
+endif()
```

Reply:
But if alpaka is installed, `alpaka_USE_MDSPAN` can only be `SYSTEM;OFF`, and those cases are handled here. What problem does your suggestion solve?

Reply:
No, you can also install mdspan separately and then set `alpaka_USE_MDSPAN=ON` that you can use the external installation.